fstest/fstests/fstests.go

```diff
@@ -1205,6 +1206,46 @@ func Run(t *testing.T, opt *Opt) {
 				}, fs.GetModifyWindow(ctx, f))
 			})
 
+			// TestFsListRootedSubdir tests putting and listing with an Fs that is rooted at a subdirectory 2 levels down
+			TestFsListRootedSubdir := func(t *testing.T) {
```

Comment:
Can you cut this test down the the minimum possible to reproduce the bug? Remember that this test gets run on 100 different backends in the integration tests!

Reply:
I can. The idea was to also look for other similar errors that could conceivably pop up in the future, but maybe some of these are already covered by other tests.

Reply:
Made this change.